pkg/controller/daemon/daemoncontroller.go

```diff
@@ -899,11 +924,16 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *v1.Node, ds *exten
 		return false, false, false, err
 	}
 
+	// Check if need to fire InsufficientResourceError
+	fireInsufficientResourceError := shouldFireInsufficientResourceError(reasons)
+
 	for _, r := range reasons {
 		glog.V(4).Infof("DaemonSet Predicates failed on node %s for ds '%s/%s' for reason: %v", node.Name, ds.ObjectMeta.Namespace, ds.ObjectMeta.Name, r.GetReason())
 		switch reason := r.(type) {
 		case *predicates.InsufficientResourceError:
-			dsc.eventRecorder.Eventf(ds, v1.EventTypeWarning, FailedPlacementReason, "failed to place pod on %q: %s", node.ObjectMeta.Name, reason.Error())
+			if fireInsufficientResourceError {
+				dsc.eventRecorder.Eventf(ds, v1.EventTypeWarning, FailedPlacementReason, "failed to place pod on %q: %s", node.ObjectMeta.Name, reason.Error())
+			}
 			shouldSchedule = false
```

Comment:
> Move the `InsufficientResourceError` case to after `PredicateFailureError` may not work if the `Predicates` returns both `InsufficientResourceError` and `ErrNodeSelectorNotMatch` as the `InsufficientResourceError` will be the error in front of `ErrNodeSelectorNotMatch` and the `InsufficientResourceError` will get fired even though we already got `ErrNodeSelectorNotMatch` error.

I'm not sure if I understand this part.  If the Predicates returns both `InsufficientResourceError` and `ErrNodeSelectorNotMatch`, we'll just hit the `ErrNodeSelectorNotMatch` and if we return immediately we're done. We won't emit "failed to place pod" event, isn't this what we want?


> Also not quite get what does `Why not just return false, false, false for intentional PredicateFailureError` mean? I think we already return `wantToRun, shouldSchedule, shouldContinueRunning = false, false, false` for `intentional` reason?

Note that we didn't return immediately here, but continue the loop. If we just want to skip emitting "failed to place pod" event when there's an intentional error, why not just return immediately?